### PR TITLE
Avoid non-capturing groups in bash regexps

### PR DESCRIPTION
**Pull request by diox**
Hopefully my last attempt at fixing https://github.com/mozilla/extension-workshop/issues/1236

### CODE REVIEW

**Comment by diox** [comment]
```bash
if [[ "2022.03.10-1-stage" =~ ^20[0-9]{2}\.[0-9]{2}\.[0-9]{2}(-[0-9]+)?-stage$ ]] ; then echo 'ok!'; else echo 'fail'; fi
ok!
if [[ "2022.03.10-stage" =~ ^20[0-9]{2}\.[0-9]{2}\.[0-9]{2}(-[0-9]+)?-stage$ ]] ; then echo 'ok!'; else echo 'fail'; fi
ok!
```

And the opposite for non-stage tags:
```bash
if [[ "2022.03.10" =~ ^20[0-9]{2}\.[0-9]{2}\.[0-9]{2}(-[0-9]+)?-stage$ ]] ; then echo 'fail!'; else echo 'ok!'; fi
ok!
if [[ "2022.03.10-1" =~ ^20[0-9]{2}\.[0-9]{2}\.[0-9]{2}(-[0-9]+)?-stage$ ]] ; then echo 'fail!'; else echo 'ok!'; fi
ok!
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -162,7 +164,7 @@ workflows:
             tags:
               # match tags such as "2022.03.08" or "2022.03.08-1", with or
               # without a -stage suffix.
-              only: /^20\d{2}\.\d{2}\.\d{2}(?:-\d+)?(?:-stage)?$/
+              only: /^20\d{2}\.\d{2}\.\d{2}(-\d+)?(-stage)?$/
```
Comment (eviljeff) [inline .circleci/config.yml:167]: these last two still have `\d` in them?

**Review by eviljeff — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (diox) [reply .circleci/config.yml:167]: Yes, I debated changing them... it's fine in circleci config...

Comment (diox) [reply .circleci/config.yml]: https://github.com/mozilla/extension-workshop/pull/1258/commits/7aeb2403051d367518993084bd9aac6231f8dd55

File: .circleci/config.yml [hunk 2]
```diff
@@ -102,6 +102,8 @@ jobs:
             ./.utils/circleci/invalidate-cloudfront-cache.sh
 
   # tags matching specific format are deployed to either stage or prod.
+  # note that batch regexp support is limited: it doesn't support non-capturing
+  # groups or [0-9] character class.
```
Comment (eviljeff) [inline .circleci/config.yml:106]: find&replace fail? 😆

Comment (diox) [reply .circleci/config.yml:106]: a013dc260c448b55f8ddb7bd3be1b90e96c35e97

**Pull request merged**